### PR TITLE
Strict type check for multiple providers

### PR DESCRIPTION
**Pull request by mlgruby**
Strict type check for multiple providers. Part of: #9708
cc: @kaxil @mik-laj tried covering all smaller provider, can I request for review?

Only big ones are left (Microsft, AWS and Google) and also we need to do strict type check for `dbapi_hook.py` as lots of 
providers are extending this one and there are functions whose types/signature is not defined.      

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by mlgruby** [comment]
42.599590 amazon
52.078754 microsoft
69.100860 google
75.000000 postgres
77.619050 databricks
80.625000 ftp
82.449500 cncf
83.333300 odbc
83.333300 openfaas
87.421875 apache
87.500000 presto ==========> from 0 to 87
87.500000 redis
88.333340 mysql
88.888900 papermill
90.178575 slack ===========> from 77 to 90
90.909100 elasticsearch
94.532960 qubole ==========> from 0 to 94
96.428575 salesforce
96.875000 mongo
98.076900 imap
100.000000 celery
100.000000 cloudant
100.000000 datadog
100.000000 dingding =======> from 0 to 100
100.000000 discord
100.000000 docker
100.000000 exasol
100.000000 facebook
100.000000 grpc
100.000000 hashicorp
100.000000 http
100.000000 jdbc
100.000000 jenkins
100.000000 jira
100.000000 opsgenie =======> from 0 to 100
100.000000 oracle
100.000000 pagerduty
100.000000 plexus
100.000000 samba ==========> from 0 to 100
100.000000 segment
100.000000 sendgrid
100.000000 sftp
100.000000 singularity
100.000000 snowflake
100.000000 sqlite
100.000000 ssh
100.000000 vertica
100.000000 yandex
100.000000 zendesk ========> from 0 to 100

File: airflow/providers/dingding/hooks/dingding.py [hunk 1]
```diff
@@ -49,20 +51,20 @@ class DingdingHook(HttpHook):
     def __init__(
         self,
         dingding_conn_id='dingding_default',
-        message_type='text',
-        message=None,
-        at_mobiles=None,
-        at_all=False,
+        message_type: str = 'text',
+        message: Union[str, dict, None] = None,
```
Comment (mik-laj) [inline airflow/providers/dingding/hooks/dingding.py]:
```suggestion
        message: Optional[Union[str, dict]] = None,
```
We mark optional arguments in this way. I think it's worth keeping one style.

File: airflow/providers/qubole/operators/qubole_check.py [hunk 2]
```diff
@@ -163,12 +164,12 @@ class QuboleValueCheckOperator(ValueCheckOperator, QuboleOperator):
     def __init__(
         self,
         *,
-        pass_value,
-        tolerance=None,
+        pass_value: Union[str, int, float],
+        tolerance: Union[int, float, None] = None,
```
Comment (mik-laj) [inline airflow/providers/qubole/operators/qubole_check.py]:
```suggestion
        tolerance: Optional[Union[int, float]] = None,
```

**Review by mik-laj — APPROVED** [review]

**Comment by mik-laj** [comment]
I am very glad that you are working on typing coverage for vendors. This will make life easier for many users as type problems can be detected much earlier.

**Review by kaxil — APPROVED** [review]
🎉

**Pull request merged**